### ISSUE TITLE
Show a year dropdown with `showMonthYearPicker`

### ISSUE THREAD

**Issue by AdrianMrn** [Hacker0x01/react-datepicker#2905]
**Is your feature request related to a problem? Please describe.**
We've got a DatePicker with `showMonthYearPicker` enabled. The expected year can be anywhere in the last 60 years, which makes it very annoying having to click the left/right buttons instead of having a year dropdown.

**Describe the solution you'd like**
I'd like to be able to add a prop that allows me to click the year, which opens a dropdown to select a different year.

Basically `showMonthYearDropdown` but without the days.

**Describe alternatives you've considered**
A worse alternative would be to have a separate month and year picker.

**Additional context**
If there is more interest in this (eg https://github.com/Hacker0x01/react-datepicker/issues/1762) and you'd be OK with adding this to the project, I'd happily submit a PR that enables this.

**Comment by AdrianMrn** [comment]
Nevermind, I just figured out there is a `renderCustomHeader` prop, so I did this:

```js
import range from 'lodash/range';

function MonthYearPicker() {
    return (
        <DatePicker
            showMonthYearPicker
            renderCustomHeader={({ date, changeYear }) => CustomYearDropdownPicker({ date, changeYear })}
        />
    );
}

function CustomYearDropdownPicker({ date, changeYear }) {
    return (
        <div className="p-1">
            <SelectInput
                isClearable={false}
                value={date.getFullYear()}
                options={range(1920, 2051).map((y) => ({ label: y.toString(), value: y }))}
                onChange={changeYear}
            />
        </div>
    );
}
```
